lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,37 @@
+module Semian
+  class AtomicEnum < AtomicInteger #:nodoc:
+    undef :increment_by
+    undef :increment
+
+    module AtomicEnumSharedImplementation
+      def initialize(name, permissions, symbol_list)
+        super(name, permissions)
+        initialize_lookup(symbol_list)
```

Comment:
So something like
```ruby
@sym_to_num = Hash[symbol_list.each_with_index.to_a]
@num_to_sym = @sym_to_num.invert
```
? I'd still like to keep it in a separate function though, since the SysV version will definitely reuse it